### PR TITLE
#160577499 Reading statistics

### PR DESCRIPTION
**Pull request by gloriaodipo**
**What does this PR do?**
- It implements functionality for displaying view count and comment count as reading statistics for articles

**Description of task to be completed**
- Users should be able to see their reading statistics

**How should this be manually tested?**
Clone the GitHub repository: git clone https://github.com/andela/ah-shakas on your machine

Navigate to the project folder: cd ah-shakas

Checkout out to this PR's branch: git checkout ft-reading-stats-160577499

Create a virtual environment using pipenv then activate using the command pipenv shell

Install dependancies: pipenv install

To run tests, run the command python manage.py test

To run the app on Postman, run the command python manage.py runserver

Start by creating a user: send a `POST` request to the endpoint `/api/users/` as below:
```
 {
    "user": {
                "username": "username", 
                 "email": "email@example.com",
                "password": "password123"
             }
}
```
- Login the user using email and password as in the above format, using the endpoint `/api/users/login/`.
- The returned JSON will include a token which will be required for creating an article.
- Create an article using the format below:
```
 {
    "article": {
                  "title": "my title", 
                  "description": "My description",
                  "body": "This is the body"
              }
}
```
- Send a `GET` request to the endpoint `/api/articles/<slug>` to get an article
-  Send a `POST` request to the endpoint `/api/articles/<slug>/comments/` to post a comment as below:
```
{
    "comment":{
        "body": "this is a comment"
    }
}
```
-  Send a `GET` request to the endpoint `/api/articles/statistics' 

- Notice the returned json displays view_count and comment_count with a value that is higher than the previous count by one`

**Screenshot**
image

**Relevant Pivotal Tracker story**
[160577499](https://www.pivotaltracker.com/n/projects/2198533/stories/160577499)

### CODE REVIEW

**Comment by lukorito** [comment]
@gloriaodipo Can you add more description to the readme? like add details about login and creating articles

File: authors/apps/articles/urls.py [hunk 1]
```diff
@@ -12,5 +12,7 @@
     path('articles/<slug>/rate/', views.RatingDetails.as_view(), name='ratings'),
     path('articles/<slug>/favourite', views.FavouriteGenericAPIView.as_view(), name="favourite"),
     path('tags/', views.TagsView.as_view(), name='tags'),
-    path('articles/<slug>/like/', views.ArticlesLikesDislikes.as_view(),  name='article-like')
+    path('articles/<slug>/like/', views.ArticlesLikesDislikes.as_view(),  name='article-like'),
+    path('articles/statistics/', views.ArticleStatsView.as_view(), name='stats'),
```
Comment (IncredibleBlackMan) [inline authors/apps/articles/urls.py]: Hey @gloriaodipo, kindly drop one of these paths, it appears to be repeated. Thank you.

**Review by IncredibleBlackMan — DISMISSED** [review]

**Comment by IncredibleBlackMan** [comment]
Hey @gloriaodipo, I have reviewed the PR and you did some really good work. I have also tested the code from my end and it works fine. Just make the requested change and this would be ready for a merge.

File: authors/apps/articles/urls.py (hunk 1 above)
Comment (gloriaodipo) [reply authors/apps/articles/urls.py]: Okay, thanks for the correction

**Review by koechkevin — COMMENTED** [review]
If the changes requested by @lukorito and @Ramon3162 are implemented, this PR should be ready for a merge

**Review by lukorito — APPROVED** [review]
I have tested this PR and it is ready for a merge

**Pull request merged**